### PR TITLE
Fixes #16101 - Added Metabase owner ingestion

### LINKED ISSUE #16101: Metabase ingestion support for owner metadata and usage type
### Discussed in https://github.com/open-metadata/OpenMetadata/discussions/16045

<div type='discussions-op-text'>

<sup>Originally posted by **julien-alpaca** April 25, 2024</sup>
@ulixius9 Any chance we can have the owner field populated in the metadata ingestion of Metabase? I believe we could use the creator of the dashboard as the owner.
Also Is there any plan to add a ingestion of type usage for Metabase?

Thank you</div>

### PR DESCRIPTION
**Pull request by SumanMaharana**
### Describe your changes:

Fixes #16101  - Added Metabase owner ingestion 

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py [hunk 1]
```diff
@@ -108,7 +109,9 @@ def get_dashboard_name(self, dashboard: MetabaseDashboard) -> str:
         """
         return dashboard.name
 
-    def get_dashboard_details(self, dashboard: MetabaseDashboard) -> dict:
+    def get_dashboard_details(
+        self, dashboard: MetabaseDashboard
+    ) -> MetabaseDashboardDetails:
```
Comment (IceS2) [inline ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:114]: This should be `Optional[MetabaseDashboardDetails]` as per the `client.get_dashboard_details` method.

File: ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py [hunk 2]
```diff
@@ -136,6 +139,24 @@ def get_project_name(self, dashboard_details: Any) -> Optional[str]:
...
+        self, dashboard_details: MetabaseDashboardDetails
+    ) -> Optional[EntityReference]:
+        """
+        Get dashboard owner from email
+        """
+        try:
+            if dashboard_details.creator_id:
+                owner_details = self.client.get_user_details(
+                    dashboard_details.creator_id
+                )
```
Comment (IceS2) [inline ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:152]: Did you forget to add the changes made to the MetabaseClient? I'm not seeing a `get_user_details` method.

Comment (SumanMaharana) [reply ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:152]: ohh yes .Thanks for pointing that out

File: ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py (hunk 1 above)
Comment (SumanMaharana) [reply ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py:114]: will do the change

**Review by IceS2 — APPROVED** [review]

**Review by pmbrull — APPROVED** [review]

**Pull request merged**